src/pages/resources.js

```diff
@@ -32,55 +32,87 @@ class Resources extends Component {
+		this.changeUrl(resourcesAndFilters[2]);
 
-			for (const resource of this.state.constantResources) {
-				if (resource.name.toLowerCase().includes(query)) {
-					resourcesThatMatchQuery.push(resource);
-				}
-			}
+		this.setState({
+			resources: resourcesAndFilters[0],
+			filterResources: resourcesAndFilters[1],
+			filterPrice: resourcesAndFilters[2]
+		});
 
-			if (!resourcesThatMatchQuery.length) {
-				this.setState({status: "No matches found..."});
-			}
+		const noMatchesFound = resourcesAndFilters[0].length !== 0;
 
-			this.setState({
-				status: null,
-				resources: resourcesThatMatchQuery
-			});
+		if (noMatchesFound) {
+			this.setState({ status: "No matches found..." });
 		}
 	};
 
-	filterResources = async event => {
-		let resources = this.state.constantResources;
-		const value = event.target.value;
-
-		if (value !== "Show All") {
-			resources = await resources.filter(resource => resource.category.toLowerCase() === value);
-			this.props.router.push(`/resources?category=${value}`, undefined, { shallow: true });
-		} else {
-			this.props.router.push("/resources", undefined, { shallow: true });
-		}
+	getResourcesAndFilters() {
+		const resources = this.state.constantResources;
+
+		const categoryFilter = document.getElementsByClassName(
+			"resources__filter-category"
+		)[0].value;
+		const priceFilter = document.getElementsByClassName(
+			"resources__filter-price"
+		)[0].value;
+		const searchFilter = document.getElementsByClassName(
+			"resources__search-filter"
+		)[0].value;
+
+		const resourcesAndFilters = [
+			resources,
+			categoryFilter,
+			priceFilter,
+			searchFilter
+		];
+
+		return resourcesAndFilters;
+	}
 
-		this.setState({resources, filterResources: value});
+	getFilteredResources = (
+		theResources,
+		theCategoryFilter,
+		thePriceFilter,
+		theSearchFilter
+	) => {
+		const showAll = "Show All";
+		const blank = "";
+
+		return theResources.filter(resource => {
+			const isResource = theCategoryFilter === showAll
+				? true
+				: resource.category.toLowerCase() === theCategoryFilter;
+
+			const isPrice = thePriceFilter === showAll
+				? true
+				: resource.free === JSON.parse(thePriceFilter);
+
+			const isSearch = theSearchFilter === blank
+				? true
+				: resource.name.toLowerCase().includes(theSearchFilter.toLowerCase());
+
+			return isPrice && isResource && isSearch;
+		});
 	};
 
-	filterPrice = async event => {
-		let resources = this.state.constantResources;
-		const value = event.target.value;
+	changeUrl(theCategoryFilter) {
+		const showAll = "Show All";
 
-		if (value !== "Show All" && ["true", "false"].includes(value)) {
-			resources = await resources.filter(resource => resource.free === JSON.parse(value));
+		if (theCategoryFilter !== showAll) {
+			this.props.router.push(
+				`/resources?category=${theCategoryFilter}`,
+				undefined,
+				{ shallow: true }
+			);
+		} else {
+			this.props.router.push("/resources", undefined, { shallow: true });
 		}
```

Comment:
You could shorten this whole section:
```suggestion
			this.props.router.push(
				`/resources${theCategoryFilter ? `?category=${theCategoryFilter}` : ""}`,
				undefined,
				{ shallow: true }
			);
```